pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -480,9 +480,9 @@ func validateMachine(machine machinev1.Machine, providerSpec machinev1.GCPMachin
 }
 
 // Returns true if machine exists.
-func (r *Reconciler) exists() (bool, error) {
+func (r *Reconciler) exists() (bool, bool, error) {
```

Comment:
What is the new bool value representing? If this is just for the invalid zone, perhaps using an invalidmachineconfiguration error would be better served?

Reply:
We need to know if, in this case the zone, doesn't exist. So we use that boolean to return the value to us so we can distinguish that case from regular errors.
And yes, the naming you are suggesting makes more sense.